File: android/src/main/kotlin/me/yohom/amapbase/map/MapFunctionRegistry.kt
package me.yohom.amapbase.map

import me.yohom.amapbase.map.handler.calculatetool.ConvertCoordinate
import me.yohom.amapbase.map.handler.createmap.*
import me.yohom.amapbase.map.handler.draw.AddMarker
import me.yohom.amapbase.map.handler.draw.AddMarkers
import me.yohom.amapbase.map.handler.draw.AddPolyline
import me.yohom.amapbase.map.handler.draw.ClearMarker
import me.yohom.amapbase.map.handler.interact.SetMapStatusLimits
import me.yohom.amapbase.map.handler.interact.SetPosition
import me.yohom.amapbase.map.handler.interact.SetZoomLevel

val MAP_METHOD_HANDLER: Map<String, MapMethodHandler> = mapOf(
        "map#setMyLocationStyle" to SetMyLocationStyle,
        "map#setUiSettings" to SetUiSettings,
        "marker#addMarker" to AddMarker,
        "marker#addMarkers" to AddMarkers,
        "marker#clear" to ClearMarker,
        "map#showIndoorMap" to ShowIndoorMap,
        "map#setMapType" to SetMapType,
        "map#setLanguage" to SetLanguage,
        "map#clear" to ClearMap,
        "map#setZoomLevel" to SetZoomLevel,
        "map#setPosition" to SetPosition,
        "map#setMapStatusLimits" to SetMapStatusLimits,
        "tool#convertCoordinate" to ConvertCoordinate,
        "offline#openOfflineManager" to OpenOfflineManager,
        "map#addPolyline" to AddPolyline
)

File: ios/Classes/map/MapFunctionRegistry.m
//
// Created by Yohom Bao on 2018-12-08.
//

#import "MapFunctionRegistry.h"
#import "MapMethodHandler.h"
#import "ClearMap.h"
#import "SetMyLocationStyle.h"
#import "SetUiSettings.h"
#import "AMapSearchKit.h"
#import "AddMarker.h"
#import "AddMarkers.h"
#import "ShowIndoorMap.h"
#import "SetMapType.h"
#import "SetLanguage.h"
#import "ClearMarker.h"
#import "SetZoomLevel.h"
#import "SetPosition.h"
#import "SetMapStatusLimits.h"
#import "ConvertCoordinate.h"
#import "OpenOfflineManager.h"
#import "AddPolyline.h"

static NSDictionary<NSString *, NSObject <MapMethodHandler> *> *_map;

@implementation MapFunctionRegistry {
}

+ (NSDictionary<NSString *, NSObject <MapMethodHandler> *> *)mapMethodHandler {
    if (!_map) {
        _map = @{
                @"map#clear": [ClearMap alloc],
                @"map#setMyLocationStyle": [SetMyLocationStyle alloc],
                @"map#setUiSettings": [SetUiSettings alloc],
                @"marker#addMarker": [AddMarker alloc],
                @"marker#addMarkers": [AddMarkers alloc],
                @"map#showIndoorMap": [ShowIndoorMap alloc],
                @"map#setMapType": [SetMapType alloc],
                @"map#setLanguage": [SetLanguage alloc],
                @"marker#clear": [ClearMarker alloc],
                @"map#setZoomLevel": [SetZoomLevel alloc],
                @"map#setPosition": [SetPosition alloc],
                @"map#setMapStatusLimits": [SetMapStatusLimits alloc],
                @"tool#convertCoordinate": [ConvertCoordinate alloc],
                @"offline#openOfflineManager": [OpenOfflineManager alloc],
                @"map#addPolyline": [AddPolyline alloc],
        };
    }
    return _map;
}

@end